sdks/v4-sdk/src/PositionManager.ts

```diff
@@ -276,25 +284,30 @@ export abstract class V4PositionManager {
 
     // If migrating, we need to settle and sweep both currencies individually
     if (isMint(options) && options.migrate) {
-      // payer is v4 positiion manager
+      if (options.useNative) {
+        planner.addUnwrap(OPEN_DELTA)
+      }
+      // payer is v4 position manager
       planner.addSettle(position.pool.currency0, false)
       planner.addSettle(position.pool.currency1, false)
       planner.addSweep(position.pool.currency0, options.recipient)
       planner.addSweep(position.pool.currency1, options.recipient)
+      // sweep wrapped native for good measure
+      if (options.useNative) {
+        planner.addSweep(position.pool.currency0.wrapped, options.recipient)
+      }
     } else {
       // need to settle both currencies when minting / adding liquidity (user is the payer)
       planner.addSettlePair(position.pool.currency0, position.pool.currency1)
     }
 
     // Any sweeping must happen after the settling.
     let value: string = toHex(0)
-    if (options.useNative) {
-      invariant(position.pool.currency0.isNative || position.pool.currency1.isNative, NO_NATIVE)
-      let nativeCurrency: Currency = position.pool.currency0.isNative
-        ? position.pool.currency0
-        : position.pool.currency1
-      value = position.pool.currency0.isNative ? toHex(amount0Max) : toHex(amount1Max)
-      planner.addSweep(nativeCurrency, MSG_SENDER)
+    // When not migrating and adding native currency, add a final sweep
+    if (options.useNative && !(isMint(options) && options.migrate)) {
```

Comment:
just dpuble triple confirming, in the migrate eth case we do not need to sweep eth since we are only unwrapping and sending in the entire contract balance so there is never any eth leftover... any edge cases here?

Reply:
okay ya ur right we just got some gnarly if else statements going on

Reply:
because now we need another if else statement at the end on whether to sweep currency0 or currency0.wrapped